K2Bridge/Visitors/RangeVisitor.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System.Globalization;
+    using K2Bridge.Models.Request.Aggregations;
+    using K2Bridge.Models.Response;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(RangeAggregation rangeAggregation)
+        {
+            Ensure.IsNotNull(rangeAggregation, nameof(TermsAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
+
+            // Start the union operator
+            rangeAggregation.KustoQL += "union ";
+
+            // Query expressions for each range
+            // (_data | where foo >= 1 and bar < 10 | summarize ['3']=avg(baz), count() | extend ['2'] = '1-10')
+            foreach (var range in rangeAggregation.Ranges)
+            {
+                rangeAggregation.KustoQL += $"(_data | {KustoQLOperators.Where} ";
+
+                var rangeClause = new Models.Request.Queries.RangeClause()
+                {
+                    FieldName = rangeAggregation.Field,
+                    GTEValue = range.From?.ToString() ?? "*",
+                    LTValue = range.To?.ToString() ?? "*",
+                };
+                rangeClause.Accept(this);
+
+                if (string.IsNullOrEmpty(rangeClause.KustoQL))
+                {
+                    // This is then "open" range, -infinity to +infinity
+                    rangeAggregation.KustoQL += "true";
+                }
+                else
+                {
+                    rangeAggregation.KustoQL += $"{rangeClause.KustoQL}";
+                }
+
+                // Insert the KustoQL generated by the metrics aggregations
+                // Followed by the default aggregation metric
+                rangeAggregation.KustoQL += $" | {KustoQLOperators.Summarize} {rangeAggregation.MetricsKustoQL}{rangeAggregation.Metric}";
```

Comment:
so here, you have the MetricsKustoQL which was set up in AggregationVisitor L33, and you use it.
This line is not easy to read. 
a. not sure what the last part means rangeAggregation.Metric
b. the name MetricsKustoQL is not very self explanatory.
c. I would add a lot more comments on this specific flow

Reply:
I have added more comments here and in `AggregationVisitor` also. Hope it makes sense.

Reply:
I know it works, but I am not sure why. 

shouldnt it be:

rangeAggregation.KustoQL += $" | {KustoQLOperators.Summarize} {rangeClause.MetricsKustoQL}{rangeAggregation.Metric}";

notice the 'rangeClause'

Reply:
The `rangeClause` is inserted in the Kusto query in line 56. In line 63, we insert the summarize, then the list of metrics (avg, sum) and finally the mandatory count().

Reply:
I have renamed `MetricsKustoQL` to `SubAggregationsKustoQL`. Hope it makes more sense.